From aa5215a0b3f8febfe9eda2ab586d9bf8ba647c7c Mon Sep 17 00:00:00 2001
From: Nicolas Fella <nicolas.fella@gmx.de>
Date: Mon, 11 Jan 2021 16:03:31 +0100
Subject: [PATCH] [applets/digitalclock] Fix showing agenda view when enabling
 plugins

The agenda view is only shown when at least one event source is active.
When going from 0 to 1 active sources the view wasn't updated correctly.

This was because the visibility of rightColumn was bound to the
visibility of its child, but since the visibility of the parent affects
the visibility of its childred (https://doc.qt.io/qt-5/qml-qtquick-item.html#visible-prop) that resulted in a broken binding.

Bind to the source property directly to avoid this

BUG: 431433
(cherry picked from commit 56064d80ff2e798460967b28243c8117ee6ebaec)
---
 applets/digital-clock/package/contents/ui/CalendarView.qml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/applets/digital-clock/package/contents/ui/CalendarView.qml b/applets/digital-clock/package/contents/ui/CalendarView.qml
index a7104100e..1541a3501 100644
--- a/applets/digital-clock/package/contents/ui/CalendarView.qml
+++ b/applets/digital-clock/package/contents/ui/CalendarView.qml
@@ -145,7 +145,7 @@ PlasmaComponents3.Page {
         ColumnLayout {
             id: rightColumn
 
-            visible: agenda.visible || worldClocks.visible
+            visible: calendar.showAgenda || calendar.showClocks
 
             Layout.minimumWidth: units.gridUnit * 14
 
-- 
GitLab

